### PR TITLE
Implement Method#===

### PR DESCRIPTION
**Pull request by havenwood**
The six Method#=== specs that were failing with `jt test :ruby25 spec/ruby/core/method` are now all passing.

### CODE REVIEW

File: src/main/ruby/core/method.rb [hunk 1]
```diff
@@ -21,7 +21,9 @@ def inspect
   def curry(curried_arity = nil)
     self.to_proc.curry(curried_arity)
   end
-  
+
```
Comment (chrisseaton) [inline src/main/ruby/core/method.rb]: We don't usually like to commit white-space changes unless you're fixing something egregious. Could I get you to un-stage this change, please?

**Comment by chrisseaton** [comment]
@eregon again should this have something to do with specs in it?

**Comment by eregon** [comment]
Can you add `spec/ruby/core/method/case_compare_spec.rb` to the `:ruby25` list in `spec/truffle.mspec`?
This is being documented in #947.

File: src/main/ruby/core/method.rb [hunk 2]
```diff
@@ -24,4 +24,6 @@ def curry(curried_arity = nil)
   
   alias_method :to_s, :inspect
 
+  alias_method :===, :call
```
Comment (eregon) [inline src/main/ruby/core/method.rb]: This is nice, but usually we try to keep aliases together for clarity.

Could you move this alias in the Java part in:
https://github.com/graalvm/truffleruby/blob/38f41992386db0500cde96bde98c3a44b37f9f5f/src/main/java/org/truffleruby/core/proc/ProcNodes.java#L213
?

Comment (havenwood) [reply src/main/ruby/core/method.rb]: Much nicer, will do!

Comment (eregon) [reply src/main/ruby/core/method.rb]: I pointed you to the wrong place (Proc instead of Method), but you found the right place anyway :)

**Pull request merged**